### PR TITLE
feat: Watson Assistant (Conversation) IAM credentials migration

### PR DESCRIPTION
**Pull request by christiancompton**

### CODE REVIEW

**Review by dingwilson — APPROVED** [review]
LGTM

File: app/index.js [hunk 1]
```diff
@@ -528,8 +528,8 @@ module.exports = Generator.extend({
         if (answers.services.indexOf('AppID') !== -1) {
           this._addService('auth', generateServiceName(this.appname, 'AppID'))
         }
-        if (answers.services.indexOf('Watson Conversation') !== -1) {
-          this._addService('conversation', generateServiceName(this.appname, 'WatsonConversation'))
+        if (answers.services.indexOf('Watson Assistant, formerly Conversation') !== -1) {
+          this._addService('conversation', generateServiceName(this.appname, 'WatsonAssistant'))
```
Comment (EnriqueL8) [inline app/index.js:532]: Would it make sense to also change the `ServiceType` to `watsonAssistant`?

File: app/index.js [hunk 2]
```diff
@@ -609,7 +609,7 @@ module.exports = Generator.extend({
           case 'objectStorage': return 'Object Storage'
           case 'auth': return 'AppID'
           case 'autoscaling': return 'Auto-scaling'
-          case 'conversation': return 'Watson Conversation'
+          case 'conversation': return 'Watson Assistant'
```
Comment (EnriqueL8) [inline app/index.js:612]: if the above is done, then this needs changing

File: app/index.js [hunk 3]
```diff
@@ -850,28 +850,26 @@ module.exports = Generator.extend({
...
-        { name: 'watsonConversationName', message: 'Enter name (blank for default):' },
-        { name: 'watsonConversationUsername', message: 'Enter username (blank for none):' },
-        { name: 'watsonConversationPassword', message: 'Enter password:', type: 'password' },
-        { name: 'watsonConversationUrl', message: 'Enter url (blank for none):' }
+        { name: 'watsonAssistantName', message: 'Enter name (blank for default):' },
+        { name: 'watsonAssistantAPIKey', message: 'Enter API key (blank for none):' },
+        { name: 'watsonAssistantUrl', message: 'Enter url (blank for default):' }
       ]
       return this.prompt(prompts).then((answers) => {
         var watsonService = this.services.conversation
```
Comment (EnriqueL8) [inline app/index.js:866]: Also this if proposal is accepted above

File: lib/helpers.js [hunk 4]
```diff
@@ -470,11 +470,10 @@ function sanitizeCredentialsAndFillInDefaults (serviceType, service) {
         tenantId: service.tenantId || ''
       }
     case 'conversation':
```
Comment (EnriqueL8) [inline lib/helpers.js:472]: And here if proposal is accepted

File: test/unit/helpers.js [hunk 5]
```diff
@@ -370,7 +368,7 @@ describe('Unit tests for helpers', function () {
       assert.equal(helpers.getBluemixServiceLabel('auth'), 'AppID')
     })
 
-    it('get label for watsonconversation', function () {
+    it('get label for watsonassistant', function () {
       assert.equal(helpers.getBluemixServiceLabel('conversation'), 'conversation')
```
Comment (EnriqueL8) [inline test/unit/helpers.js]: Wait - does the bluemix object give the `conversation` tag?

File: app/index.js (hunk 1 above)
Comment (christiancompton) [reply app/index.js:532]: Unfortunately,  when invoked by scaffolder/starter kit flow in the bluemix object, the label will be "conversation" - we need to make the standalone/prompting mode conform to this convention in serviceType.  This is not a part of the rebrand.

Comment (EnriqueL8) [reply app/index.js:532]: oh sad times...

**Pull request merged**